unblob/cli.py

```diff
@@ -263,6 +272,11 @@ def cli(
     extra_dir_handlers = plugin_manager.load_dir_handlers_from_plugins()
     dir_handlers += tuple(extra_dir_handlers)
 
+    if clear_skip_magics and skip_magic is not DEFAULT_SKIP_MAGIC:
+        skip_magic = tuple(sorted(set(skip_magic).difference(DEFAULT_SKIP_MAGIC)))
+    else:
+        skip_magic = tuple(sorted(set(skip_magic).union(DEFAULT_SKIP_MAGIC)))
+
```

Comment:
`difference` is a bit stronger than needed: we just do not want to add the defaults, but when added by a user, we do not want to remove them.
```suggestion
    extra_magics_to_skip = () if clear_skip_magics else DEFAULT_SKIP_MAGIC
    skip_magic = tuple(sorted(set(skip_magic).union(extra_magics_to_skip)))
```

Reply:
Thanks. I actually struggled with this one, my brain was a bit foggy ^^

Reply:
Tests are failing if I implement your change. This is due to the fact that regardless of `clear_skip_magics`, `skip_magic` defaults to `DEFAULT_SKIP_MAGIC` if the user does not explicitly provide others.

From my perspective, using `--clear-skip-magics` without any `--skip-magic` means the default skip_magic should be cleared. See the test at https://github.com/onekey-sec/unblob/pull/693/files#diff-4e8715c7a425ee52e74b7df4d34efd32e8c92f3e60bd51bc2e1ad5943b82032eR386.